Kernel/Net/LocalSocket.cpp

```diff
@@ -399,14 +399,27 @@ ErrorOr<void> LocalSocket::getsockopt(OpenFileDescription& description, int leve
 
     MutexLocker locker(mutex());
 
+    size_t buffer_size;
     socklen_t size;
     TRY(copy_from_user(&size, value_size.unsafe_userspace_ptr()));
 
     switch (option) {
     case SO_SNDBUF:
-        return ENOTSUP;
+        if (size < sizeof(size_t))
+            return EINVAL;
+        buffer_size = m_for_server.ptr()->space_for_writing();
```

Comment:
This seems to be the currently available write space, not the entire buffer's size? So not quite correct. Also, you can access all members with a direct arrow operator, e.g. `m_for_server->whatever_method_i_need()`